examples/MQ/pixelDetector/src/PixelDigiBinSource.cxx

```diff
@@ -108,22 +99,22 @@ Int_t PixelDigiBinSource::ReadEvent(UInt_t i)
     fRunId = head[0];
     fMCEntryNo = head[1];
     fPartNo = head[2];
-    fEventHeader->SetRunId(fRunId);
-    fEventHeader->SetMCEntryNumber(fMCEntryNo);
-    fEventHeader->SetPartNo(fPartNo);
+    fEventHeader.SetRunId(fRunId);
+    fEventHeader.SetMCEntryNumber(fMCEntryNo);
+    fEventHeader.SetPartNo(fPartNo);
 
     for (Int_t idata = 0; idata < head[3]; idata++) {
         LOG(debug) << "    --/" << idata << "/-->    " << dataCont[idata * dataSize + 0] << " / "
                    << dataCont[idata * dataSize + 1] << " / " << dataCont[idata * dataSize + 2] << " / "
                    << dataCont[idata * dataSize + 3] << " / "
                    << " 0.";
-        new ((*fDigis)[fNDigis]) PixelDigi(-1,
-                                           (Int_t)dataCont[idata * dataSize + 0],
-                                           (Int_t)dataCont[idata * dataSize + 1],
-                                           (Int_t)dataCont[idata * dataSize + 2],
-                                           (Int_t)dataCont[idata * dataSize + 3],
-                                           0.,
-                                           0.);
+        new (fDigis[fNDigis]) PixelDigi(-1,
+                                        (Int_t)dataCont[idata * dataSize + 0],
```

Comment:
I would propose C++ cast instead of C-style casts when the code is cleaned up anyway.

Reply:
Thanks for noticing!

I don't think, a cast is needed at all. This is just a short int, that becomes an int. That should not even envolve any losses.

Reply:
Ok. I did not check the data types but upcasting should be fine. Downcasting is a problem and would end in a compiler error.
So the solution would be to simply remove the casts.

Is there a clang-tidy option to warn about C-style casts?

Reply:
I removed the casts.

> Is there a clang-tidy option to warn about C-style casts?

Yes, there is: [cppcoreguidelines-pro-type-cstyle-cast](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-cstyle-cast.html)

But enabling it would spam us with hundreds of warnings. Nobody will fix them in the near future. If they're in headers, then they will be even more often.

Our current clang-tidy setup has issues with too many warnings (it misses sometimes some, sometimes even strange new ones appear). This is basically, because we compile in parallel and the warnings interleaf.

Reply:
I wouldn't switch them on by default. But maybe one should run clang-tidy before the final pull request.